AutoCollection/HttpRequests.ts

```diff
@@ -90,26 +94,35 @@ class AutoCollectHttpRequests {
                         onRequest(request, response);
                     }
                 });
-            });
+            }
         }
 
-        var originalHttpsServer = https.createServer;
-        https.createServer = (options, onRequest) => {
-            return originalHttpsServer(options, (request:http.ServerRequest, response:http.ServerResponse) => {
-                // Set up correlation context
-                var requestParser = new HttpRequestParser(request);
-                var correlationContext = this._generateCorrelationContext(requestParser);
+        function wrapServerEventHandler(server) {
+            const originalOn = server.on.bind(server);
+            server.on = (eventType: string, eventHandler) => {
```

Comment:
Our unit tests are flagging type errors around these for having implicit `any`. Can you use their concrete types?

Reply:
Sorry for not checking the tests before pushing. Hope I catched all cases now.